### PR TITLE
[Merged by Bors] - Set max and min pool connection size

### PR DESCRIPTION
**Pull request by acrrd**

### CODE REVIEW

**Review by janpetschexain — COMMENTED** [review]
the error message in the failing tests seem to indicate that the default `max_pool_size` is too constraining:
```
ERROR xayn_web_api::storage::postgres::client: error=pool timed out while waiting for an open connection
```

File: web-api-shared/src/postgres.rs [hunk 1]
```diff
@@ -53,8 +53,14 @@ pub struct Config {
     /// If true skips running db migrations on start up.
     skip_migrations: bool,
 
-    /// Number of connections in the pool.
-    pub min_pool_size: u8,
+    /// Minimum number of connections in the pool.
+    /// When the pool is built, this many connections will be automatically spun up.
+    /// This value is clamped internally to not exceed `max_pool_size`.
```
[the remark this answers is not in the document]
Comment (acrrd) [reply web-api-shared/src/postgres.rs:58]: the constraint is not needed because we clamp. The clamping is done by sqlx.

**Comment by acrrd** [comment]
> the error message in the failing tests seem to indicate that the default `max_pool_size` is too constraining:
> 
> ```
> ERROR xayn_web_api::storage::postgres::client: error=pool timed out while waiting for an open connection
> ```

The problem is that the min was too high for the local deployment. I modified it to have min as 0 instead.

File: web-api-shared/src/postgres.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (janpetschexain) [reply web-api-shared/src/postgres.rs:58]: i see, it's very deeply neested in the sqlx code and took me some time to find it, but with the comment/docs it should be sufficient.

File: web-api-shared/src/postgres.rs [hunk 2]
```diff
@@ -53,8 +53,15 @@ pub struct Config {
     /// If true skips running db migrations on start up.
     skip_migrations: bool,
 
-    /// Number of connections in the pool.
+    /// Minimum number of connections in the pool.
+    /// When the pool is built, this many connections will be automatically spun up.
+    /// This value is clamped internally to not exceed `max_pool_size`.
+    /// Default: 0
```
Comment (janpetschexain) [inline web-api-shared/src/postgres.rs:59]: ❓ should we document the defaults in the comment? this is likely to be forgotten if we change the defaults. and anyone interested in the defaults can just jump to the `Default` impl.

same question for the `max_pool_size`.

**Review by janpetschexain — APPROVED** [review]
lgtm

File: web-api-shared/src/postgres.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (acrrd) [reply web-api-shared/src/postgres.rs:58]: `/// This value is clamped internally to not exceed `max_pool_size`.` This line is copied from their doc of `min_connections`.

File: web-api-shared/src/postgres.rs (hunk 2 above)
Comment (acrrd) [reply web-api-shared/src/postgres.rs]: removed

**Comment by acrrd** [comment]
bors merge

**Pull request closed without merging**